.github/instructions/copilot-instructions.md

```diff
@@ -0,0 +1,124 @@
+---
+applyTo: '**'
+---
+
+## Must Do
+
+- When writing **integration tests**, refer to the existing tests under the `tests/harvester_e2e_tests/integrations` directory.
+- When writing **API tests**, refer to the existing tests under the `tests/harvester_e2e_tests/apis` directory.
+- If you mention "shared in the class" or "sharing in the class", make sure the `@pytest.fixture` scope is set to `class`. Use a `yield` statement to define the shared fixture.
+- Each test case **must include at least one passing and one failing scenario**.
+- **All comments must be written in English** — do not use other languages.
+
+### Must Do for Harvester API
+
+- When adding a new Harvester API and the corresponding resource **does not already exist** under `apiclient/harvester_api/managers`, create a new file with the **plural form** of the resource name.
+    - For example, if the resource is "image", search for `apiclient/harvester_api/managers/image.py` or `apiclient/harvester_api/managers/images.py`.
+- Load the new resource in `apiclient/harvester_api/api.py` and register it in `apiclient/harvester_api/managers/__init__.py`.
+- Refer to `apiclient/harvester_api/managers/images.py` as a sample implementation.
+- Follow these standard status codes to validate Harvester API responses:
+  - `201` → Resource successfully created
+  - `200` → Operation succeeded (general case)
+
+## Don't Do
+
+- Do **not** use method names with `_` prefixes in tests, such as `api_client.[resource]._[method]`.
+- Do **not** define a function inside another function, for example:
```

Comment:
for example as https://github.com/harvester/tests/blob/71d70d9ddcae47f3c1eefb57be3450341629d33b/harvester_e2e_tests/fixtures/virtualmachines.py#L119-L368

so it would not be **Don't**, but need more consideration.

Reply:
Okay. I didn't notice we have this pattern, thanks!